### ISSUE TITLE
In-built support for obtaining a GitHub App token is failing with `A JSON web token could not be decoded`

### ISSUE THREAD

**Issue by rtyley** [scala-steward-org/scala-steward-action#565]
PR https://github.com/scala-steward-org/scala-steward-action/pull/428 back in December 2022 introduced built-in support for generating the GitHub App token from the `github-app-key` parameter, meaning that it was no longer necessary to have a separate `Generate token` step in the GitHub workflow ✨ 

Unfortunately, the in-built support in `scala-steward-action` is now failing. The `A JSON web token could not be decoded` error is consistent, and looks like [this](https://github.com/guardian/scala-steward-public-repos/actions/runs/8095568959/job/22122550693#step:5:23):

```
Run scala-steward-org/scala-steward-action@v2.63.0
✓ Default Scala Steward configuration set to: common-config/scala-steward.conf
✓ Using multiple repos file: REPOSITORIES.md
Error:  ✕ A JSON web token could not be decoded
```

It looks like this _isn't_ happening for Steward workflows that don't use the in-built support - for instance, the [TypeLevel Steward ](https://github.com/typelevel/steward/actions) [is still using `tibdex/github-app-token`](https://github.com/typelevel/steward/blob/dd21fad9a748dd6acc827e2fb243b11ec09dd6a5/.github/workflows/steward.yml#L19), and it's working:

<img width="1643" alt="image" src="https://github.com/scala-steward-org/scala-steward-action/assets/52038/b468ee8d-0ea2-4355-aabf-b90c9dde2175">

cc @alejandrohdezma

**Comment by alejandrohdezma** [comment]
Hey @rtyley, this is quite weird. I also use the `github-app-key` in my own Scala Steward instance and [I just triggered a run](https://github.com/alejandrohdezma/.github/actions/runs/8100711251) and everything went fine.

**Comment by alejandrohdezma** [comment]
Can you try updating to latest version (`2.64.0`)? I merged a bunch of dependabot updates including ones for the auth-app and requests libraries that might help on this issue.

**Comment by mzuehlke** [comment]
Can it be that the key expired ? or got revoked ?
The error message is not generated inside this action.... just an idea.

**Comment by rtyley** [comment]
> Can it be that the key expired ? or got revoked ?

YES. Oh, this is embarrassing. When I was doing debug on https://github.com/scala-steward-org/scala-steward/issues/3300 yesterday, I needed to run [Scala Steward locally to attach a debugger, while using 'live' settings](https://github.com/guardian/scala-steward-public-repos/blob/main/running-locally.md) - which involved downloading a fresh secret key for our Steward GitHub App. I noticed there was what looked like superfluous key, so I deleted it - but obviously, it was not superfluous.

I have generated & stored a new private key, and the Scala Steward run has started up perfectly - so the problem was entirely my misconfiguration, and nothing to do with `scala-steward-action` at all.

Thank you @mzuehlke & @alejandrohdezma for helping me debug the problem, and apologies for taking up your valuable time - I really appreciate it.
